File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        char[][] board = new char[n][m];

        for(int i=0; i<n; i++) {
            String s = br.readLine();
            for(int j=0; j<m; j++) {
                board[i][j] = s.charAt(j);
            }
        }

        int count = 0;
        int min = Integer.MAX_VALUE;
        for(int i=0; i<=n-8; i++) {
            for(int j=0; j<=m-8; j++) {
                count = 0;
                char c = 'W';
                for(int p=i; p<i+8; p++) {
                    for(int q=j; q<j+8; q++) {
                        if(board[p][q] != c)
                            count++;

                        c = c == 'W' ? 'B' : 'W';
                    }
                    c = c == 'W' ? 'B' : 'W';
                }
                count = Math.min(count, 64-count);
                min = Math.min(min, count);
            }
        }

        System.out.println(min);
    }
}

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.kt
import java.io.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val (n, m) = br.readLine().split(" ").map { it.toInt() }
    val board = Array(n) { br.readLine().toCharArray() }
    var minPaint = Int.MAX_VALUE
    
    for (i in 0..(n - 8)) {
        for (j in 0..(m - 8)) {
            var countW = 0 // 'W'로 시작
            var countB = 0 // 'B'로 시작
            
            for (x in 0 until 8) {
                for (y in 0 until 8) {
                    val current = board[i + x][j + y]
                    val expectedW = if ((x + y) % 2 == 0) 'W' else 'B'
                    val expectedB = if ((x + y) % 2 == 0) 'B' else 'W'
                    if (current != expectedW) countW++
                    if (current != expectedB) countB++
                }
            }
            minPaint = minOf(minPaint, countW, countB)
        }
    }
    println(minPaint)
}